### PR TITLE
Custom event processor validation and tests

### PR DESCRIPTION
**Pull request by jdrake**
Adds:
- Custom event processor validation and tests
- Files to get CI working in this repo

### CODE REVIEW

File: fluxio_parser/visitors/script.py [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
-from ..util import parse_options
-from .event_processor import EventProcessorVisitor
-from .state_machine import StateMachineVisitor
-from .task import TaskVisitor
+from ..exceptions import assert_supported_operation, UnsupportedOperation  # noqa
+from ..resource_decorators import RESOURCE_DECORATOR_MAP  # noqa
+from ..util import parse_options  # noqa
+from .event_processor import EventProcessorVisitor  # noqa
+from .state_machine import StateMachineVisitor  # noqa
+from .task import TaskVisitor  # noqa
```
Comment (ssantana-ns) [inline fluxio_parser/visitors/script.py:11]: does the linter not like `..`? Also any reason you use that over the fully qualified path?

**Review by ssantana-ns — APPROVED** [review]
just one question but not a change inducing one

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - id: check-builtin-literals
+      - id: check-byte-order-marker
+      - id: check-executables-have-shebangs
+        # Need to define stages explicitly since `default_stages` was not being respected
+        stages: [commit]
+      - id: check-merge-conflict
+      - id: debug-statements
+      - id: forbid-new-submodules
+      - id: no-commit-to-branch
+      - id: trailing-whitespace
```
Comment (ns-bdesimone) [inline .pre-commit-config.yaml:23]: can you add the new line hook that i added to talos?

**Review by ns-bdesimone — APPROVED** [review]
One small request doesnt block a merge though

File: fluxio_parser/visitors/script.py (hunk 1 above)
Comment (jdrake) [reply fluxio_parser/visitors/script.py:11]: The linter is ok with `..` but the sort order plugin wants to put `.` before `..`. Reason for relative links is mostly a personal preference. I might do a follow up and replace everything w absolute.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jdrake) [reply .pre-commit-config.yaml:23]: Yeah sure, i can add in a follow up

**Pull request merged**